internal/integrationtest/compile_1/compile_test.go

```diff
@@ -1321,3 +1321,46 @@ func buildWithCustomBuildPathAndOUtputDirFlag(t *testing.T, env *integrationtest
 		require.NotEmpty(t, content)
 	}
 }
+
+func TestCompileWithOutputDirFlagIgnoringFoldersCreatedByEsp32RainMakerLib(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("update")
+	require.NoError(t, err)
+
+	// Update index with esp32 core and install it
+	url := "https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json"
+	_, _, err = cli.Run("core", "update-index", "--additional-urls="+url)
+	require.NoError(t, err)
+	_, _, err = cli.Run("core", "install", "esp32:esp32@3.0.7", "--additional-urls="+url)
+	require.NoError(t, err)
+
+	sketchName := "RainMakerSketch"
+	sketchPath := cli.SketchbookDir().Join(sketchName)
+	_, _, err = cli.Run("sketch", "new", sketchPath.String())
+	require.NoError(t, err)
+
+	// When importing the `RMaker` library, the esp32 core produces in output a folder,
+	// containing all the binaries exported. The name of the folder matches the sketch name.
+	// In out case:
+	// cache-dir/
+	//  |- RainMakerSketch.ino/ <--- This should be ignored
+	//  |- RainMakerSketch.ino.uf2
+	//  |- RainMakerSketch.ino.bin
+	//  |- ...
+	err = sketchPath.Join(sketchName + ".ino").WriteFile([]byte(`
+		#include "RMaker.h"
```

Comment:
I was wondering if it would be better if we simulate the outcome of this test without relying on this library.

If we manually create the `RainMakerSketch.ino` folder inside the build path, it would be more future-proof. We don't risk breaking this test in case espressif changes something to that library. (Without downloading new core the test would be faster)

@cmaglie any preference on this?

Reply:
> If we manually create the RainMakerSketch.ino folder inside the build path, it would be more future-proof.

The problem is that you must create the folder after the compile but before the export and those actions are quickly performed one after the other on the same "compile" command.
Maybe a better way to test is to extract an `exportSketchFiles` function from the compile command and unit-test it.

BTW I'm fine also with this test.

Reply:
@cmaglie since we're using the build path we have 2 options:
- precreate the folder -> compile -> assertions
- compile -> create the folder -> compile -> assertions